src/components/InterviewsTable.js

```diff
@@ -0,0 +1,336 @@
+import React from "react";
+import Table from "react-bootstrap/Table";
+import Modal from "react-bootstrap/es/Modal";
+import { Row } from "react-bootstrap";
+import Col from "react-bootstrap/es/Col";
+import { library } from "@fortawesome/fontawesome-svg-core";
+import { faEye, faEdit } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import Form from "react-bootstrap/es/Form";
+import Button from "react-bootstrap/es/Button";
+library.add(faEye);
+library.add(faEdit);
+
+
+class CustomTable extends React.Component {
+    state = {
+        showProfile: false,
+        editProfile: false,
+        status: "",
+        comment: "",
+        validated: false,
+        meetingRoom: "",
+        interviewer: "",
+        date: ""
+    };
+
+    updateInterview = e => {
+        const form = e.currentTarget;
+        if (form.checkValidity() === false) {
+            e.preventDefault();
+            e.stopPropagation();
+        } else {
+            e.preventDefault();
+            e.stopPropagation();
+            let formData = JSON.stringify(this.state);
+            fetch( {
+                headers: {
+                    "Content-Type": "application/json"
+                },
+                method: "PUT",
+                body: formData
+            });
+        }
+
+        this.setState({ validated: true });
+    };
+
+    render() {
+        const head = [
+            "#",
+            "Candidate",
+            "Interviewer",
+            "Date",
+            "Time",
+            "Location",
+            "CV",
+            ...(this.props.editInterview ? ["Edit"] : [])
+        ];
+        return (
+            <React.Fragment>
+                {this.state.selectedInterview && (
+                    <Modal
+                        size="lg"
+                        show={this.state.showInterview}
+                        onHide={() => this.setState({ showInterview: false })}
+                        aria-labelledby="example-modal-sizes-title-lg"
+                    >
+                        <Modal.Header closeButton>
+                            <Modal.Title id="example-modal-sizes-title-lg">
+                                {this.state.selectedInterview[0]["candidate"]}
+                            </Modal.Title>
+                        </Modal.Header>
+                        <Modal.Body>
+                            <Row>
+                                <Col>
+                                    <h5>Candidate Info</h5>
+                                    <Table striped bordered hover>
+                                        <tbody>
+                                        <tr>
+                                            <td>Candidate</td>
```

Comment:
Maybe deconstructed this.state.selectedInterview[0] will make the code cleaner and maintainable